nmaipy/parcels.py

```diff
@@ -429,163 +437,135 @@ def feature_attributes(
 
         # Select and produce results for the primary feature of each feature class
         if class_id in CLASSES_WITH_PRIMARY_FEATURE:
```

Comment:
@mbewley Does the pipeline calculate defensible space for each building or only the primary? We'll get our best retro results if we do all buildings and then aggregate in a few different ways. Often we find the most lift on the lowest score per parcel (highest risk building on the parcel).

Reply:
Only the primary, because it gets messy with edges. Recommendation is to set primary as "nearest" a representative point from the desired building, so that gets used. That way if there are other buildings nearby, it doesn't try to compute them (they will be near the parcel boundary and get only partially valid buffers, and if we extend the AOI to them, more buildings will be included, etc.).

This means that if you want every roof in a parcel, you construct an nmaipy query file with one row per roof.